core/src/main/java/hudson/lifecycle/UnixLifecycle.java

```diff
@@ -80,8 +80,9 @@ public void restart() throws IOException, InterruptedException {
             LIBC.fcntl(i, F_SETFD,flags| FD_CLOEXEC);
         }
 
-        // exec to self
+        // exec to self, except for a possible daemon-supplied prefix (JENKINS-67075):
         String exe = args.get(0);
+        exe = exe.replaceFirst("^(.*): /", "/");
```

Comment:
Should this regular expression discard the first string that does not contain a ':' character, ending with a ':'?  Shouldn't it require at least one character in the prefix string before the ':' character?
```suggestion
        exe = exe.replaceFirst("^([^:]+): /", "/");
```

I still need to test this on a Debian 11 system before I'm ready to approve the pull request or my tentatively proposed change to the pull request.

Reply:
You're absolutely right; I was so focused on figuring out how replacements work in the Java world that I have lost track of the actual pattern, and forgot to try and improve it. Given https://sources.debian.org/src/adduser/3.118/adduser/#L866 (and probably other constraints on different platforms), it looks like an “interesting” topic.

Anyway, it looks like:
 - `*` → `+` makes sense;
 - excluding `:` from the character class looks good to me;
 - maybe excluding ` ` (space) and/or `/` as well for good measure?